### DISCUSSION TITLE
Easier to get started in FastAPI than Starlite?

### DISCUSSION THREAD

**Discussion by metaperl** [litestar-org/litestar#1183] [Q&A] [answered]
Would anyone care to comment on [this post](https://www.reddit.com/r/Python/comments/10ypnnn/comment/j7z2pzy/?utm_source=share&utm_medium=web2x&context=3) where someone claimed that 

> I tried starlite and found it to be very not friendly. I can create a Pydantic model that’s is my route’s function input and then create a model for the output and setup a calculation route that pairs input to output in a matter of seconds. I spent a day trying to make a basic equivalent in starlite and found there to be a lot of overhead and things not working as simply as FastAPI. I would love to find an alternative since FastAPI has long term maintenance issues and will go by the wayside

**Comment by provinzkraut** [comment, marked as answer]
Thanks for bringing this up, it's always good to know what people struggle with.

I'd say that from a certain perspective, yes, it can be easier to get started with FastAPI than with Starlite. It might also be the other way around for some. 

Currently FastAPI *does* have a fairly exhaustive tutorial aimed at people with basically zero prior knowledge, so if that's you, yes, you're probably going to have an easier time. I'm currently trying to address this, but as it turns out, writing good quality tutorials on that level isn't all that easy, and, credit where it's due, Mr. Ramirez of FastAPI is really good at that.

On the other hand, many things are arguably *easier to do* in Starlite than in FastAPI, so if you're on a level where you already know exactly what you want to do, you might have an easier time with Starlite. Although we can probably also improve in that area, by making our documentation easier to navigate. (@Jarmos-san is currently working on that!).

**Comment by jarmos-san** [comment]
I've used FastAPI for production projects for a while now & I can attest, Starlite will be equally (if not more) easy to use as it is for FastAPI! Sure there's a slight syntax change here & there to get used to. For example the [Class-based Controllers](https://starlite-api.github.io/starlite/latest/reference/controller.html#starlite.controller.Controller) (_see this "[Minimal Example](https://starlite-api.github.io/starlite/latest/index.html#minimal-example)" for more reference on usage_) differs from the traditional approach other "easier-to-use" frameworks like FastAPI or Flask used.

FastAPI also relies on another framework & provide a thin wrapper API around it. And in doing so I've had to look at its source code a couple of times because the documentations were lacking a specific explanation. @Goldziher & @peterschutt has put significant effort into designing Starlite from the ground-up. The results of which are:

1. Starlite is faster than FastAPI, Starlette & friends (see [the benchmarks](https://starlite-api.github.io/starlite/latest/benchmarks.html)).
2. The public API is also very featureful & way more Pythonic.

What's lacking is proper, easy-to-read & referable documentations. There's active work to improve it (_see the "[documentations overhaul](https://github.com/orgs/starlite-api/projects/1)" project board_) & I'm working on improving the "Usage Guidelines" of the documentations as well. You can find the first PR of the many future ones there will be w.r.t to this context here - #1153. @provinzkraut is also working on drafting a tutorial application built using Starlite. These tasks are time-consuming but we hope our efforts in revamping the documentations will help everyone get started with using Starlite easier (even by a bit).

And the other area where I strongly believe the community can help in making Starlite easier to use is by:

1. Joining our [Discord server](https://discord.gg/xsySNKSUcW), opening Discussion threads on GitHub **(very recommended)** & so on for us to make it easier to respond to your usage queries.
2. Share feedback on areas where Starlite can improve on. And if you plan on using it in a production environment do let us know of the concerns you might have?
3. Create content (could be YT videos or blog posts) showcasing Starlite, its use cases, tutorials & much more. As mentioned in the previous comment, Ramirez excels in this aspect but Starlite is a community-maintained project & only the community efforts can make the project easier to use.

That said, there's significant work going on to release a v2 of the project soon! So expect a ton of things to change with the second major release of the project. Do provide us with feedback so that we can continuously make the project better for the community & the project's user base.

**Comment by jet10000** [comment]
I like starlite, it's more systematic
